execsrc/Main.hs

```diff
@@ -53,39 +73,45 @@ argumentParser = info (arguments <**> helper)
                  )
 
 -- | The actual entry point of the application
-application :: FilePath -> FilePath -> EvalM ()
-application ss dirOfModelSolutions = do
+application :: Gen String -> FilePath -> FilePath -> EvalM ()
+application gen ss dirOfModelSolutions = do
```

Comment:
This section has too many nested `do`s, `case`s, and parts should be split into functions.